src/main/java/org/opensearch/sdk/NettyTransport.java

```diff
@@ -40,9 +40,23 @@
 
 public class NettyTransport {
     private static final String NODE_NAME_SETTING = "node.name";
+    private ExtensionsRunner extensionsRunner;
     private final TransportInterceptor NOOP_TRANSPORT_INTERCEPTOR = new TransportInterceptor() {
     };
 
+    /**
+     *
+     * @param extensionsRunner method to call
+     */
+    public NettyTransport(ExtensionsRunner extensionsRunner) {
+        this.extensionsRunner = extensionsRunner;
+    }
+
+    /**
+     * empty constructor for getNetty4Transport method call
+     */
+    public NettyTransport() {}
```

Comment:
Why is this needed?  Without initializing the `extensionsRunner` you'll end up with an NPE when you try to initialize.

Reply:
Before I added this empty constructor in `NettyTransport.java`, The test class `TestNetty4Transport` and `TransportCommunicationIT` shows error message "The constructor NettyTransport() is undefined".

Reply:
The constructor problem is fixed.  Thank @vibrantvarun Varun help me on that.